src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -86,11 +100,13 @@
 
     private final ConcurrentMap<Memtable, MemtableIndex> liveMemtableIndexMap;
```

Comment:
We might want to consider wrapping up `liveMemtableIndexMap` and the things that touch it in a `MemtableIndexManager`. This would be similar to `IndexViewManager`, but whatever the case, it feels a little odd that we search and index Memtables directly on the context. WDYT?